sycl/doc/extensions/proposed/sycl_ext_oneapi_group_load_store.asciidoc

```diff
@@ -0,0 +1,596 @@
+* `InputT` must be convertible to value type of `OutputIteratorT`.
+* `InputT` and value type of `OutputIteratorT` must be trivially copyable
+ and default constructible.
+
+_Effects_: Stores single element `in` to `out_iter` by using the `g` group
+object to identify memory location as `out_iter` + `g.get_local_linear_id()`
+
+Properties may provide a hint for optimization purposes
+xref:optimization_hints[optimization_hints].
+
+==== `sycl::vec` Overload
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+// Store API sycl::vec overload
+template<typename Group, typename InputT, int N,
+         typename OutputIteratorT, typename Properties = empty_properties_t>
+void group_store(Group g, const sycl::vec<InputT, N>& in,
+                      OutputIteratorT out_iter, Properties = {});
+}
+----
+
+_Preconditions_: `out_iter` must be the same for all work-items
+in the group.
+
+._Constraints_:
+* Only available if `Group` is a work-group or sub-group.
+* `OutputIteratorT` suppose to be a random access iterator.
+* `InputT` must be convertible to value type of `OutputIteratorT`.
+* `InputT` and value type of `OutputIteratorT` must be trivially copyable
+ and default constructible.
+
+_Effects_: Stores `N` elements from `in` vec to `out_iter`
+using the `g` group object.
+Properties may specify xref:data_placement[data placement].
+Default placement is a blocked one:
+`out_iter[g.get_local_linear_id() * N + i]` = `in[i];`
+in striped case:
+`out_iter[g.get_local_linear_id() + g.get_local_linear_range() * i]` = `in[i];`
+for `i` between `0` and `N`.
+Properties may also provide a hint for optimization purposes
+xref:optimization_hints[optimization_hints].
+
+==== Fixed-size Array Overload
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+// Store API
+template<typename Group, typename OutputIteratorT, typename InputT,
+        std::size_t ElementsPerWorkItem, typename Properties = empty_properties_t>
+void group_store(Group g, sycl::span<InputT, ItemsPerWorkItem> in,
+                 OutputIteratorT out_iter,  Properties = {});
+
+}
+----
+
+._Preconditions_:
+* `out_iter` must be the same for all work-items in the group.
+
+._Constraints_:
+* Only available if `Group` is a
+work-group or sub-group.
+* `OutputIteratorT` suppose to be a random access iterator.
+* `InputT` must be convertible to value type of `OutputIteratorT`.
+* `InputT` and value type of `OutputIteratorT` must be trivially copyable
+ and default constructible.
+
+_Effects_: Stores `ElementsPerWorkItem` elements from `in` span to `out_iter`
+using the `g` group object.
+
+Properties may specify xref:data_placement[data placement].
+Default placement is a blocked one:
+`out_iter[g.get_local_linear_id() * ItemsPerWorkItem + i]` = `in[i];`
+in striped case:
+`out_iter[g.get_local_linear_id() + g.get_local_linear_range() * i]` = `in[i];`
+for `i` between `0` and `ItemsPerWorkItem`.
+Properties may also provide a hint for optimization purposes
```

Comment:
```suggestion
Properties may also provide assertions which can enable better optimizations
```

Or, it might read better to integrate the link like this:

```suggestion
Properties may also provide xref:optimization_properties[assertions] which can enable better optimizations.
```

And similar for the other place that have this sentence.